sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2079,4 +2078,601 @@ case class GpuConv(num: Expression, fromBase: Expression, toBase: Expression)
   override def third: Expression = toBase
 
   override def dataType: DataType = StringType
-}
\ No newline at end of file
+}
+
+case class GpuFormatNumber(x: Expression, d: Expression)
+    extends GpuBinaryExpression with ExpectsInputTypes with NullIntolerant {
+  
+  override def left: Expression = x
+  override def right: Expression = d
+  override def dataType: DataType = StringType
+  override def nullable: Boolean = true
+  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType)
+
+  private def stripLeadingZeros(cv: ColumnVector): ColumnVector = {
+    val (isZero, zeroCv) = withResource(Scalar.fromString("0")) { zero =>
+      (cv.startsWith(zero), ColumnVector.fromScalar(zero, cv.getRowCount.toInt))
```

Comment:
if `fromScalar` fails, like with an OOM the result of startsWith will be leaked.  To make this work we would need to rewrite it as.

```
  val (isZero, zeroCv) = withResource(Scalar.fromString("0")) { zero =>
    closeOnExcept(cv.startsWith(zero)) { isZero =>
      (isZero, ColumnVector.fromScalar(zero, cv.getRowCount.toInt))
    }
  }
```

Reply:
thanks, done.